cpp/src/hash/md5_hash.cu

```diff
@@ -342,8 +276,14 @@ std::unique_ptr<column> md5_hash(table_view const& input,
       }
     });
 
-  return make_strings_column(
-    input.num_rows(), std::move(offsets_column), std::move(chars_column), 0, std::move(null_mask));
+  // Build an output null mask from the logical AND of all input columns' null masks.
+  auto [null_mask, null_count] = cudf::detail::bitmask_and(input, stream);
```

Comment:
This changes the null behavior so that a hash of null input data returns a null value, rather than the hash of an empty byte string. This ensures that users can distinguish between the hash of empty bytes and the hash of null inputs.

Reply:
I retracted this change in behavior because it caused tests to fail. We do need to fix this so that null inputs produce null outputs, and I will open a separate PR to fix it.